test/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoMockTest.java

```diff
@@ -125,43 +124,44 @@ public void getSubpopulationConstructsCriteriaIfNotSaved() {
     }
     
     @Test
-    public void getSubpopulationsForUserRetrievesCriteria() {
-        CriteriaContext context = createContext();
-        
-        List<Subpopulation> subpops = dao.getSubpopulationsForUser(context);
-        Subpopulation subpop = subpops.get(0);
-        Criteria criteria = subpop.getCriteria();
-        assertEquals(CRITERIA, criteria);
+    public void physicalDeleteSubpopulationDeletesCriteria() {
+        dao.deleteSubpopulation(TEST_STUDY, SUBPOP_GUID, true, false);
         
-        verify(criteriaDao).getCriteria(criteria.getKey());
-        verifyNoMoreInteractions(criteriaDao);
+        verify(criteriaDao).deleteCriteria(createSubpopulation().getCriteria().getKey());
     }
-
+    
     @Test
-    public void getSubpopulationsForUserConstructsCriteriaIfNotSaved() {
-        when(criteriaDao.getCriteria(any())).thenReturn(null);
-        CriteriaContext context = createContext();
-        
-        List<Subpopulation> subpops = dao.getSubpopulationsForUser(context);
-        Subpopulation subpop = subpops.get(0);
-        Criteria criteria = subpop.getCriteria();
-        assertNotNull(criteria);
+    public void logicalDeleteSubpopulationDoesNotDeleteCriteria() {
+        dao.deleteSubpopulation(TEST_STUDY, SUBPOP_GUID, false, false);
         
-        verify(criteriaDao).getCriteria(criteria.getKey());
+        verify(criteriaDao, never()).deleteCriteria(createSubpopulation().getCriteria().getKey());
     }
-
-    @Test
-    public void physicalDeleteSubpopulationDeletesCriteria() {
-        dao.deleteSubpopulation(TEST_STUDY, SUBPOP_GUID, true);
+    
+    @Test(expected = BadRequestException.class)
+    public void doNotAllowDeleteOfDefault() {
+        Subpopulation defaultSubpop = Subpopulation.create();
+        defaultSubpop.setGuid(SUBPOP_GUID);
+        defaultSubpop.setDefaultGroup(true);
         
-        verify(criteriaDao).deleteCriteria(createSubpopulation().getCriteria().getKey());
+        doReturn(defaultSubpop).when(dao).getSubpopulation(TEST_STUDY, SUBPOP_GUID);
+        doReturn(Criteria.create()).when(criteriaDao).getCriteria(any());
+        
+        dao.deleteSubpopulation(TEST_STUDY, SUBPOP_GUID, true, false);
     }
     
     @Test
-    public void logicalDeleteSubpopulationDoesNotDeleteCriteria() {
-        dao.deleteSubpopulation(TEST_STUDY, SUBPOP_GUID, false);
+    public void allowDeleteOfDefault() {
```

Comment:
Looks like this checks physical delete of default subpopulation.  Is there a case where you would do a logical delete of default?  it doesn't look like that case is covered.

Reply:
Added test that logical delete also cannot delete the default (unless you're calling from the delete everything service method which sets the flag to override this, used for tests).